### PR TITLE
MAINT make NumPy the default frontend

### LINKED ISSUE #868: Make NumPy the default frontend
Since we promised earlier:

```
  /home/jenkins/workspace/kymatio_dev/kymatio/frontend/entry.py:20: DeprecationWarning: Torch frontend is currently the default, but NumPy will become the default in the next version.
    warnings.warn("Torch frontend is currently the default, but NumPy will become the default in the next"
```

### PR DESCRIPTION
**Pull request by MuawizChaudhary**
This PR resolves #868, making Numpy the default frontend like we said we would.

### CODE REVIEW

File: tests/scattering2d/test_frontend_scattering2d.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
 from kymatio import Scattering2D
-from kymatio.scattering2d.frontend.torch_frontend import ScatteringTorch2D
+from kymatio.scattering2d.frontend.numpy_frontend import ScatteringNumPy2D
 
 # Check that the default frontend is Torch and that errors are correctly launched.
 def test_scattering2d_frontend():
     scattering = Scattering2D(2, shape=(10, 10))
-    assert isinstance(scattering, ScatteringTorch2D), 'Torch frontend is not selected by default'
+    assert isinstance(scattering, ScatteringNumPy2D), 'Torch frontend is not selected by default'
```
Comment (janden) [inline tests/scattering2d/test_frontend_scattering2d.py:9]: `Torch` → `Numpy`

**Review by janden — CHANGES_REQUESTED** [review]

**Review by janden — APPROVED** [review]

File: tests/scattering2d/test_frontend_scattering2d.py [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 from kymatio import Scattering2D
-from kymatio.scattering2d.frontend.torch_frontend import ScatteringTorch2D
+from kymatio.scattering2d.frontend.numpy_frontend import ScatteringNumPy2D
 
-# Check that the default frontend is Torch and that errors are correctly launched.
+# Check that the default frontend is Numpy and that errors are correctly launched.
 def test_scattering2d_frontend():
     scattering = Scattering2D(2, shape=(10, 10))
-    assert isinstance(scattering, ScatteringTorch2D), 'Torch frontend is not selected by default'
+    assert isinstance(scattering, ScatteringNumPy2D), 'Numpy frontend is not selected by default'
```
Comment (lostanlen) [inline tests/scattering2d/test_frontend_scattering2d.py:9]: `Numpy` --> `NumPy`

**Review by lostanlen — COMMENTED** [review]
just a nitpick about NumPy capitalization

**Review by lostanlen — APPROVED** [review]
thanks

**Pull request merged**